### PR TITLE
[v2/Windows]Set default extension for save file dialog

### LINKED ISSUE #3202: runtime.SaveFileDialog automatically  add the suffix of the filter
### Is your feature request related to a problem? Please describe.

wails:v2.7.0 
os:windows 10
When I use the Save File dialog, the filter I set does not work, and the return value will not automatically add a suffix name

### Describe the solution you'd like

Is there a way to automatically add the filter suffix I have selected when I save as a file？

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by almas1992**
# Description

According to the  [comment](https://github.com/wailsapp/wails/blob/12d633642115f077a67dac5a65180581062eb5e0/v2/internal/go-common-file-dialog/cfd/DialogConfig.go#L36-L39), the default extension should be set when filters are set, so that the filename is automatically appended with the extension.

Fixes #3202

## Type of change
  
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
  
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration using `wails doctor`.

- [x] Windows
- [ ] macOS
- [ ] Linux
  
## Test Configuration

Please paste the output of `wails doctor`. If you are unable to run this command, please describe your environment in as much detail as possible.

# Checklist:

- [x] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [ ] My code follows the general coding style of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **Bug Fixes**
	- Fixed the save file dialog not automatically appending file extensions on Windows.

### CODE REVIEW

**Comment by snopywithchary** [comment]
if I change my filter，Will the default suffix change based on the filter I select?
eg: this is my filters Images (*.png *.xpm *.jpg);;Text (*.txt)

**Comment by almas1992** [comment]
> if I change my filter，Will the default suffix change based on the filter I select? eg: this is my filters Images (*.png *.xpm _.jpg);;Text (_.txt)

If the user selects a different file filter, the default extension will be automatically updated to match the new file filter.

File: v2/internal/frontend/desktop/windows/dialog.go [hunk 1]
```diff
@@ -121,6 +121,10 @@ func (f *Frontend) SaveFileDialog(options frontend.SaveDialogOptions) (string, e
 		Folder:      defaultFolder,
 	}
 
+	if len(options.Filters) > 0 {
+		config.DefaultExtension = strings.TrimPrefix(strings.Split(options.Filters[0].Pattern, ";")[0], "*")
+	}
```
[the remark this answers is not in the document]
Comment (leaanthony) [reply v2/internal/frontend/desktop/windows/dialog.go:126]: Splitting on a non existent separator is fine if you're only interested in the first element

[the remark this answers is not in the document]
Comment (almas1992) [reply v2/internal/frontend/desktop/windows/dialog.go:126]: > Splitting on a non existent separator is fine if you're only interested in the first element

The reason I split by semicolon first is to handle patterns with multiple extensions, such as 
```go
{
    DisplayName: "Image Files (*.jpg, *.png)",
    Pattern:     "*.jpg;*.png",
}
```
Just take the first extension, i.e. `.jpg`

**Review by leaanthony — APPROVED** [review]

**Comment by leaanthony** [comment]
Thanks @almas1992 🙏

**Pull request merged**